actors/verifreg/src/lib.rs

```diff
@@ -672,13 +672,51 @@ impl Actor {
                 term_min: req.term_min,
                 term_max: req.term_max,
                 expiration: req.expiration,
-            })
+            });
+            allocation_total += DataCap::from(req.size.0);
+        }
+
+        let st: State = rt.state()?;
+        let mut claims = st.load_claims(rt.store())?;
+        let mut updated_claims = Vec::<(ClaimID, Claim)>::new();
+        for req in &reqs.extensions {
+            let provider_id = rt
+                .resolve_address(&req.provider)
+                .with_context_code(ExitCode::USR_ILLEGAL_ARGUMENT, || {
+                    format!("failed to resolve provider address {}", req.provider)
+                })?;
+            let claim = state::get_claim(&mut claims, provider_id, req.claim)?
+                .with_context_code(ExitCode::USR_NOT_FOUND, || {
+                    format!("no claim {} for provider {}", req.claim, provider_id)
+                })?;
+            let policy = rt.policy();
+
+            validate_claim_extension(req, claim, policy, curr_epoch)?;
+            updated_claims.push((req.claim, Claim { term_max: req.term_max, ..*claim }));
```

Comment:
We are going to run into mild problems with `verifreg::ExtendClaimTerms` if we don't update the client.  If its a new client extending the term here they won't be able to later max out term_max using that method.  This is probably ok because they could just do it the right way to start here + I think that that method only exists to deal with the first allocation and associated imperfections with the delegation of setting terms by market.  However its still a weird quirk we should think through.

Reply:
I considered this and it's by design. Datacap-spending clients like this should just do it up front. The extension-by-client mechanism mainly exists to address shortcomings in markets. If we did want to support the new "client" to change their mind, I think I'd still want to retain the original client in state anyway.

One possible case were we _might_ wish to have changed it is if in the future the maximum allocation term increase above 5y, and then an datacap-spending client could wish they had been allowed more in the first place.